airflow/jobs/scheduler_job_runner.py

```diff
@@ -1113,6 +1294,20 @@ def _run_scheduler_loop(self) -> None:
                     self.log.debug("Waiting for processors to finish since we're using sqlite")
                     self.processor_agent.wait_until_finished()
 
+                # This is using a new session.
+                self._end_spans_of_externally_ended_ops()
```

Comment:
I would rather we didn't use `@provide_session` here and were explicit about managing and passing in a session

Reply:
I'll move it under the new session below.